docs/tasks/access-kubernetes-api/migrate-third-party-resource.md

```diff
@@ -0,0 +1,163 @@
+    versions:
+    - name: v1
+    ```
+
+    A matching CustomResourceDefinition could look like this:
+
+    ```yaml
+    apiVersion: apiextensions.k8s.io/v1beta1
+    kind: CustomResourceDefinition
+    metadata:
+      name: crontabs.stable.example.com
+    spec:
+      scope: Namespaced
+      group: stable.example.com
+      version: v1
+      names:
+        kind: CronTab
+        plural: crontabs
+        singular: crontab
+    ```
+
+1.  **Install the CustomResourceDefinition**
+
+    While the source TPR is still active, install the matching CRD with `kubectl create`.
+    Existing TPR data will remain accessible because TPRs take precedence over CRDs when both try
+    to serve the same resource.
+
+    After you create the CRD, make sure the *Established* condition goes to True.
+    You can check it with a command like this:
+
+    ```shell
+    kubectl get crd -o 'custom-columns=NAME:{.metadata.name},ESTABLISHED:{.status.conditions[?(@.type=="Established")].status}'
+    ```
+
+    The output should look like this:
+
+    ```console
+    NAME                          ESTABLISHED
+    crontabs.stable.example.com   True
+    ```
+
+1.  **Stop all clients that use the TPR**
+
+    The API server will attempt to prevent mutation of TPR data for the resource while it
+    copies objects to the CR, but it can't guarantee consistency in all cases, such as with
+    [multiple masters](/docs/admin/high-availability/).
+    Stopping clients, such as TPR-based custom controllers, helps to avoid inconsistencies in
+    the copied data.
+
+    In addition, clients that watch TPR data will not receive any more events once the migration
+    begins.
+    You must restart them after the migration completes so they start watching CRD data instead.
+
+1.  **Back up TPR data**
+
+    In case the data migration fails, save a copy of existing data for the resource:
+
+    ```shell
+    kubectl get crontabs --all-namespaces -o yaml > crontabs.yaml
+    ```
+
+    You should also save a copy of the TPR definition if you don't have one already:
+
+    ```shell
+    kubectl get thirdpartyresource cron-tab.stable.example.com -o yaml --export > tpr.yaml
+    ```
+
+1.  **Delete the TPR definition**
+
+    Normally, when you delete a TPR definition, the API server tries to clean up any objects stored
+    in that resource.
+    Since you created a matching CRD, the server will copy objects to the CRD instead of deleting
+    them.
+
+    ```shell
+    kubectl delete thirdpartyresource cron-tab.stable.example.com
+    ```
+
+1.  **Verify the new CRD data**
+
+    It can take up to 10 seconds for the TPR controller to notice that you deleted the TPR definition
+    and initiate the migration. TPR data will remain accessible during this time.
+
+    Once the migration completes, the resource will begin serving through the CRD.
+    Check that all your objects were correctly copied:
+
+    ```shell
+    kubectl get crontabs --all-namespaces -o yaml
```

Comment:
Out of scope for this PR, but @chenopis, what do you think of eventually implementing custom Markdown like DigitalOcean has (https://www.digitalocean.com/community/tutorials/digitalocean-s-writing-guidelines#variables)? That way it's clearer that things like `crontab` are meant to be swapped out and are not part of the real command.

See:
<img width="1135" alt="screen shot 2017-06-23 at 3 43 10 pm" src="https://user-images.githubusercontent.com/25401650/27502679-c2fbe518-582a-11e7-9911-8fe109bb9559.png">